core/src/main/java/org/geysermc/geyser/network/netty/GeyserServer.java

```diff
@@ -94,19 +95,28 @@ public final class GeyserServer {
 
     private final GeyserImpl geyser;
     private EventLoopGroup group;
+    // Split childGroup may improve IO
+    private EventLoopGroup childGroup;
     private final ServerBootstrap bootstrap;
     private EventLoopGroup playerGroup;
 
     @Getter
     private final ExpiringMap<InetSocketAddress, InetSocketAddress> proxiedAddresses;
+    private final int listenCount;
 
-    private ChannelFuture bootstrapFuture;
+    private ChannelFuture[] bootstrapFutures;
 
     public GeyserServer(GeyserImpl geyser, int threadCount) {
         this.geyser = geyser;
-        this.group = TRANSPORT.eventLoopGroupFactory().apply(threadCount);
+        // bindCount = 1/2 of processors, and max to 20
+        // It's offers up to 20*250=5000 workload
+        this.listenCount = Bootstraps.isReusePortAvailable() ? Math.max(1, Math.min(20, threadCount / 4)) : 1;
```

Comment:
It might be a good idea to make this configurable.
The reason I suggest this is because some might want to use more than 50% (or 25% if the cpu has the same amount of threads as cores), or less.
The average server has 0-4 players online (ranging from 90% on Velocity to 99% on the Spigot platform, Floodgate stats), and allocating more threads for listening only adds context switching

Reply:
1+

Reply:
https://github.com/GeyserMC/Geyser/blob/master/core/src/main/java/org/geysermc/geyser/GeyserImpl.java#L354-L358
Something like this probably, with a system property?

Reply:
Or actually, threadCount is exactly that variable that's configurable there.. do we want to allow to configure the listenCount separately?

Reply:
I'd go with a system property yeah, otherwise we'll get too much config options.
I'd optimize it for the average server, so 1 listen thread. And have a description for that property on our wiki stating that this could be used for servers with a high player count (and mention a specific number like 100 players)

Reply:
Emmmm – what about setting count default to 2? It’s a good balance, right? Not too heavy on resources, but still keeps things smooth for servers that might miss tweaking this. Helps avoid those unexpected drops in connections.